lib/Module/IntrinsicCleaner.cpp

```diff
@@ -191,6 +191,66 @@ bool IntrinsicCleanerPass::runOnBasicBlock(BasicBlock &b, Module &M) {
         dirty = true;
         break;
       }
+#if LLVM_VERSION_CODE >= LLVM_VERSION(8, 0)
+      case Intrinsic::sadd_sat:
+      case Intrinsic::ssub_sat:
+      case Intrinsic::uadd_sat:
+      case Intrinsic::usub_sat: {
+        IRBuilder<> builder(ii);
+
+        Value *op1 = ii->getArgOperand(0);
+        Value *op2 = ii->getArgOperand(1);
+
+        unsigned int bw = op1->getType()->getPrimitiveSizeInBits();
+
+        Value *overflow = 0;
+        Value *result = 0;
+        Value *saturated = 0;
```

Comment:
Please use `nullptr` in the above three lines.

Reply:
Should I do the same in the overflowing case above this diff? It uses the same `Value *_= 0;` pattern

Reply:
Hm, good point. If you want to change it, you should probably do so in at least a separate commit or (even better) a separate PR (which could then also contain the renaming changes from earlier) to keep this one as concise and self-contained as possible. That said, we have a lot of code that needs some (C++-wise) modernization here, so in case you're not modifying any of this old code, it is perfectly fine to leave it as it is.

Reply:
In that case I'll leave the overflowing code as is for now. I might go on a refactoring/testing spree later, though.